stdlib/public/core/ManagedBuffer.swift

```diff
@@ -474,6 +508,31 @@ extension ManagedBufferPointer {
   @inlinable
   internal static var _elementOffset: Int {
     _onFastPath()
+    var increment: Int = 0
+    if !_swift_stdlib_has_malloc_size() {
+      increment = MemoryLayout<Int>.stride
+    }
+    let offset: Int = _headerOffset + MemoryLayout<Header>.size + increment
+    return _roundUp(
+      offset,
+      toAlignment: MemoryLayout<Element>.alignment)
```

Comment:
In the no-mallocsize case, this calculation does not take the capacity's alignment into account, so I believe it may produce incorrect results. Consider the case where `MemoryLayout<Header>.size == 1` and `MemoryLayout<Element>.alignment == 1`: 

```swift
let offset: Int = 16 + 1 + 8
_roundUp(25, toAlignment: 1) == 25
```

But the first element would be stored at offset 32.